pennylane/ops/op_math/exp.py

```diff
@@ -176,6 +176,10 @@ def _unflatten(cls, data, metadata):
 
     # pylint: disable=too-many-arguments
     def __init__(self, base, coeff=1, num_steps=None, id=None):
+        if base is not None and not isinstance(base, Operator):
+            raise TypeError(
+                f"base is expected to be of type None or Operator, but received {type(base)}"
```

Comment:
Is `base=None` supposed to be valid?

Reply:
It used to be the default, so I didn't want to disallow it, but I do think it's weird.

Reply:
i'd just remove the None check then, plus `None` isn't an instance of `Operator`

Reply:
I'm really not a big fan of that `None`. @trbromley Thoughts?

Reply:
not sure, but it was removed (fairly quietly, it seems) in #3691

Reply:
@albi3ro @mudit2812 do you have any memory of this?